### PR TITLE
Configure routing

### PR DESCRIPTION
**Pull request by mgolawski**
-- Adds `react-router-dom` for basic routing
-- Adds enzyme and set basic configuration for React testing
-- Adds global method `getRoutesMap` for testing component routing

### CODE REVIEW

File: src/components/App/App.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/* global getRoutesMap:readonly */
```
Comment (mgolawski) [inline src/components/App/App.test.js]: Declaring globals seems to be the most elegant way to share some helper logic in Jest tests. Let me know if you have any other idea for that @blurbyte.

File: src/components/App/App.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { Route } from 'react-router-dom';
+
+import Headline from './Headline';
+import Cart from '../Cart';
+import Checkout from '../Checkout';
+
+function App() {
+  return (
+    <section>
+      <Headline>Hello Khol's</Headline>
```
Comment (blurbyte) [inline src/components/App/App.js]: Let's remove real name from here.

File: src/components/App/App.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import Checkout from '../Checkout';
+
+const render = path =>
+  mount(
+    <Router initialEntries={[path]} keyLength={0}>
+      <App />
+    </Router>
+  );
+
+describe('COMPONENTS - App', () => {
```
Comment (blurbyte) [inline src/components/App/App.test.js]: Cant see any reason to add such prefix in all caps. *App* should be enough. 
Also I would rather put tests in separate directory in each of such folders. For now it is not a problem but imagine having 5+ files and 5 additional tests - could be messy. Therefore I suggest such structure:
```
App
|-- __tests__
    |-- App.test.js
|-- App.js
|-- index.js
```

File: src/components/App/App.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  it('renders self and correct routes', () => {
```
Comment (blurbyte) [inline src/components/App/App.test.js]: Could be a bit more precise: 'renders self and correct component in /cart route'

File: src/components/App/__snapshots__/App.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,233 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (blurbyte) [inline src/components/App/__snapshots__/App.test.js.snap]: I think this snapshot is a bit too noisy.
Overall we are not interested in detail what each component at given route contains.
One of the solutions could be mocking them like that:
`jest.mock('../Cart', () => 'Cart')`

File: src/components/App/__snapshots__/App.test.js.snap [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+          Object {
+            "$$typeof": Symbol(react.forward_ref),
+            "attrs": Array [],
+            "componentStyle": ComponentStyle {
+              "componentId": "sc-bdVaJa",
+              "isStatic": true,
+              "lastClassName": "iulWLw",
+              "rules": Array [
+                "
+  font-size: 2rem;
```
Comment (blurbyte) [inline src/components/App/__snapshots__/App.test.js.snap]: I think I forgot to add *jest-styled-components* (https://github.com/styled-components/jest-styled-components). Will make snapshots a bit more readable.

File: src/components/Cart/Cart.test.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Cart from './Cart';
+
+const shallowRender = () => shallow(<Cart />);
+
+describe('COMPONENTS - Cart', () => {
```
Comment (blurbyte) [inline src/components/Cart/Cart.test.js]: I think snapshot would be better here.

File: src/components/Checkout/__snapshots__/Checkout.test.js.snap [hunk 8]
```diff
@@ -0,0 +1,232 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (blurbyte) [inline src/components/Checkout/__snapshots__/Checkout.test.js.snap]: Again this snapshot is too noisy.

File: src/setupTests.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+import Adapter from 'enzyme-adapter-react-16';
```
Comment (blurbyte) [inline src/setupTests.js:1]: Probably would make more sense to put it in project root and configure point to it: `"setupTestFrameworkScriptFile": "./setupTest.js"`.
After installing *jest-styled-components* `import 'jest-styled-components'` could be added there as well.

**Review by blurbyte — COMMENTED** [review]
Good job 🥇. I would welcome a few changes thought :)

**Review by blurbyte — APPROVED** [review]
Lets merge :)

**Pull request merged**